File: source/icdab_open/icdab_open/ViewController.m
//
//  ViewController.m
//  icdab_open
//
//  Created by Faisal Memon on 07/10/2020.
//

#import "ViewController.h"
#import "objc/message.h"

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    // Do any additional setup after loading the view.
}


- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}

- (void)openFile:(NSString*)webAddress
{
    NSURL *url = [NSURL URLWithString:webAddress];
    [[NSWorkspace sharedWorkspace] openURL:url];
}

- (IBAction)openButtonAction:(id)sender {
    // Enable Strict Checking of objc_mesgSend Calls set to
    // No (non-default value) and use an Intel Mac in order
    // to get the following to compile.
    objc_msgSend(self,
                 @selector(openFile:),
                 self.fileNameTextFieldOutlet.stringValue);
}
@end

File: source/icdab_edge/icdab_edge/ViewController.swift
//
//  ViewController.swift
//  icdab_edge
//
//  Created by Faisal Memon on 09/09/2018.
//  Copyright © 2018 Faisal Memon. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    static let crash = Crash()
    
    @IBAction func crashCorruptMallocButtonPressed(_ sender: UIButton) {
        ViewController.crash.corruptMalloc()
    }
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

